plugin/src/features/Compilation/index.tsx

```diff
@@ -32,49 +34,60 @@ export const Compilation = ({ setAccordian }: CompilationProps) => {
   const remixClient = useAtomValue(remixClientAtom)
   const isValidSolidity = useAtomValue(isValidSolidityAtom)
   const currentFilename = useAtomValue(currentFilenameAtom)
+  const currentFilepath = useAtomValue(currentFilepathAtom)
   const currentWorkspacePath = useAtomValue(currentWorkspacePathAtom)
-
-  const setContracts = useSetAtom(contractsAtom)
-  const setSelectedContract = useSetAtom(selectedContractAtom)
-
   const { status, isCompiling } = useAtomValue(compilationAtom)
 
+  const setDeployStatus = useSetAtom(deployStatusAtom)
+
   const setCompileStatus = useSetAtom(compileStatusAtom)
   const setIsCompiling = useSetAtom(isCompilingAtom)
   const setCompileErrorMessages = useSetAtom(compileErrorMessagesAtom)
 
-  const solidityVersion = useAtomValue(solidityVersionAtom)
+  const [compilationType, setCompilationType] = useAtom(compilationTypeAtom)
 
-  const compilations = [
-    {
-      validation: isValidSolidity,
-      isLoading: isCompiling,
-      onClick: compile,
-      id: 1
-    }
-  ]
+  const [isContractsFolderAvailable, setIsContractsFolderAvailable] = useState(true)
+  const { emitErrorToRemix, writeResultsToArtifacts, getDefaultWorkspaceContents, setCompiledContracts } =
+    useCompileHelpers()
+
+  useEffect(() => {
+    remixClient.fileManager
+      .readdir(`${currentWorkspacePath}/`)
+      .then((workspaceFiles: any) => {
+        setIsContractsFolderAvailable(!!workspaceFiles?.contracts?.isDirectory)
+      })
+      .catch((error) => {
+        console.error(`Failed to read current workspace ${error.message}`)
+      })
+  }, [currentWorkspacePath, remixClient])
 
-  async function compile(): Promise<void> {
+  async function handleCompile({ type }: { type: CompilationType }): Promise<void> {
+    const workspaceContents = getDefaultWorkspaceContents()
+
+    setCompilationType(type)
     setIsCompiling(true)
+    setDeployStatus('IDLE')
     setCompileStatus('Compiling...')
     setCompileErrorMessages([])
-    // clear current file annotations: inline syntax error reporting
     await remixClient.editor.clearAnnotations()
     try {
-      const workspaceContents = {
-        config: {
-          version: solidityVersion,
-          user_libraries: []
-        },
-        contracts: [] as Array<{ file_name: string; file_content: string; is_contract: boolean }>
+      const allContractFiles = await getAllContractFiles(remixClient, currentWorkspacePath)
+      console.log('All contract files', allContractFiles)
```

Comment:
@satyambnsal This is debug log I guess. Print all the files. Shall be removed in prod